Exiled.API/Features/Player.cs

```diff
@@ -1534,10 +1534,10 @@ public byte GetEffectIntensity<T>()
         public ReadOnlyCollection<PlayerEffect> GetAllEffects()
         {
             ReadOnlyCollection<PlayerEffect> ActiveEffects = new ReadOnlyCollection<PlayerEffect>();
-            for (int i = 0; i < Enum.GetValue(typeof(EffectType)).GetUpperBound(0); i++)
+            foreach (var effect in ReferenceHub.playerEffectsController.AllEffects)
             {
-              if (GetEffect((EffectType) i).Intensity > 0)
-                ActiveEffects.Add(GetEffect((EffectType)i));
+              if (GetEffect((EffectType)effect.Value).Intensity > 0)
+                ActiveEffects.Add(GetEffect((EffectType)effect.Value));
```

Comment:
The `GetEffect`s here are unnecessary. Please take a look at `GetEffect`'s definition.